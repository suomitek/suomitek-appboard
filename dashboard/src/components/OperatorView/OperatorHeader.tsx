import { RouterAction } from "connected-react-router";
import * as React from "react";
import { Link } from "react-router-dom";
import { app } from "../../shared/url";

interface IOperatorHeaderProps {
  id: string;
  icon?: string;
  hideButton?: boolean;
  disableButton?: boolean;
  description: string;
  cluster: string;
  namespace: string;
  version: string;
  provider: string;
  namespaced: boolean;
  push: (location: string) => RouterAction;
}

class OperatorHeader extends React.Component<IOperatorHeaderProps> {
  public state = {
    modalIsOpen: false,
  };

  public render() {
    const {
      id,
      icon,
      description,
      cluster,
      namespace,
      version,
      provider,
      hideButton,
      disableButton,
    } = this.props;
    return (
      <header>
        <div className="ChartView__heading margin-normal row">
          <div className="col-1 ChartHeader__icon">
            <div className="ChartIcon">
              <img className="ChartIcon__img" src={icon} alt="icon" />
            </div>
          </div>
          <div className="col-9">
            <div className="title margin-l-small">
              <h1 className="margin-t-reset">{id}</h1>
              <h5 className="subtitle margin-b-normal">
                {/* TODO(andresmgot): Filter by provider */}
                <span>{version} - Provided by </span>
                <Link to={app.operators.list(cluster, namespace)}>{provider}</Link>
              </h5>
              <h5 className="subtitle margin-b-reset">{description}</h5>
            </div>
          </div>
          {!hideButton && (
            <div className="col-2 ChartHeader__button">
              <button
                className="button button-primary button-accent"
                onClick={this.redirect}
                disabled={disableButton}
              >
                {disableButton ? "Deployed" : "Deploy"}
              </button>
            </div>
          )}
        </div>
        <hr />
      </header>
    );
  }

  public redirect = () => {
    this.props.push(app.operators.new(this.props.cluster, this.props.namespace, this.props.id));
  };
}

export default OperatorHeader;
