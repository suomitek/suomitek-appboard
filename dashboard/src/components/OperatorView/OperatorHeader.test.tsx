import { shallow } from "enzyme";
import * as React from "react";
import OperatorHeader from "./OperatorHeader";

const defaultProps = {
  id: "foo",
  icon: "/path/to/icon.png",
  description: "this is a description",
  cluster: "default",
  namespace: "kubeapps",
  version: "1.0.0",
  provider: "Kubeapps",
  namespaced: false,
  push: jest.fn(),
};

it("renders the header", () => {
  const wrapper = shallow(<OperatorHeader {...defaultProps} />);
  expect(wrapper).toMatchSnapshot();
});

it("omits the button", () => {
  const wrapper = shallow(<OperatorHeader {...defaultProps} hideButton={true} />);
  expect(wrapper.find("button")).not.toExist();
});

it("disables the button", () => {
  const wrapper = shallow(<OperatorHeader {...defaultProps} disableButton={true} />);
  const button = wrapper.find("button");
  expect(button).toExist();
  expect(button.prop("disabled")).toBe(true);
  expect(button.text()).toBe("Deployed");
});
