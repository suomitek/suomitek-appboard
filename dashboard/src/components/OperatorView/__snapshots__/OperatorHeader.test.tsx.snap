// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the header 1`] = `
<header>
  <div
    className="ChartView__heading margin-normal row"
  >
    <div
      className="col-1 ChartHeader__icon"
    >
      <div
        className="ChartIcon"
      >
        <img
          alt="icon"
          className="ChartIcon__img"
          src="/path/to/icon.png"
        />
      </div>
    </div>
    <div
      className="col-9"
    >
      <div
        className="title margin-l-small"
      >
        <h1
          className="margin-t-reset"
        >
          foo
        </h1>
        <h5
          className="subtitle margin-b-normal"
        >
          <span>
            1.0.0
             - Provided by 
          </span>
          <Link
            to="/c/default/ns/kubeapps/operators"
          >
            Kubeapps
          </Link>
        </h5>
        <h5
          className="subtitle margin-b-reset"
        >
          this is a description
        </h5>
      </div>
    </div>
    <div
      className="col-2 ChartHeader__button"
    >
      <button
        className="button button-primary button-accent"
        onClick={[Function]}
      >
        Deploy
      </button>
    </div>
  </div>
  <hr />
</header>
`;
